src/Database/PostgreSQL/PQTypes/SQL/Builder.hs

```diff
@@ -408,9 +415,28 @@ instance Sqlable SqlInsertSelect where
     , emitClausesSepComma "RETURNING" $ sqlInsertSelectResult cmd
     ]
 
+checkAndRememberMaterializationSupport :: (MonadDB m, MonadIO m, MonadMask m) => m ()
+checkAndRememberMaterializationSupport = do
+  res :: Either DBException Int64 <- try . withNewConnection $ do
+    runSQL01_ $ "WITH t(n) AS MATERIALIZED (SELECT 1) SELECT n FROM t LIMIT 1"
+    fetchOne runIdentity
+  liftIO $ writeIORef withMaterializedSupported (isRight res)
+
+withMaterializedSupported :: IORef Bool
+{-# NOINLINE withMaterializedSupported #-}
+withMaterializedSupported = unsafePerformIO $ newIORef False
+
+-- This has to be a thunk because we are using the IORef reading in unsafePerformIO
```

Comment:
This shouldn't matter, AFAIK it can be a `Bool`.

Reply:
For some reason that didn't work for me - rechecking

Reply:
What if that value gets read before being properly initialized? (lazy init and stuff?)
Anyway it worked for some reason 🤔